### PR TITLE
feat: Implement Github Actions Validation Models

### LINKED ISSUE #1122: Config Validation: Github Actions - implement models required for validation
**Description**
Github Actions - implement models required for validation.

**Acceptance Criteria**
- [ ] All models required for Github Actions Source Config Validation are implemented.
- [ ] All models are tested properly.

**Additional context**
[CI Integrations Config Validator](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/04_ci_integrations_config_validator.md)

### PR DESCRIPTION
**Pull request by solid-maxim**
Fixes #1122 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/ci_integrations/lib/client/github_actions/models/workflow.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:equatable/equatable.dart';
+
+/// A class that represents a Github Actions workflow.
+class Workflow extends Equatable {
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/lib/client/github_actions/models/workflow.dart:7]:
```suggestion
class GithubActionsWorkflow extends Equatable {
```
To be consistent with other classes.

File: metrics/ci_integrations/test/client/github_actions/mappers/github_token_scope_mapper_test.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    test(
+      ".unmap() unmaps the GithubTokenScope.repo to the repo scope value",
+      () {
+        const expectedScope = GithubTokenScopeMapper.repo;
+
+        final scope = mapper.unmap(GithubTokenScope.repo);
+
+        expect(scope, equals(expectedScope));
+      },
+    );
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/github_actions/mappers/github_token_scope_mapper_test.dart:32]: Should we test that the `unmap` method returns null if the given value does not match any enum value? Also, should we test the situation when we are passing `null` to the `map` and `unmap` methods?

File: metrics/ci_integrations/test/client/github_actions/models/github_repository_test.dart [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    };
+
+    const owner = GithubUser(id: id, login: login);
+    const githubRepository = GithubRepository(id: id, name: name, owner: owner);
+
+    test(
+      "creates an instance with the given parameters",
+      () {
+        const githubRepository =
+            GithubRepository(id: id, name: name, owner: owner);
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/github_actions/models/github_repository_test.dart:30]:
```suggestion
        const githubRepository = GithubRepository(
          id: id,
          name: name,
          owner: owner,
        );
```

File: metrics/ci_integrations/test/client/github_actions/models/github_repository_test.dart [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    test(
+      ".toJson() converts an instance to the json encodable map",
+      () {
+        final json = githubRepository.toJson();
+
+        expect(json, equals(githubRepositoryJson));
+      },
+    );
+  });
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/github_actions/models/github_repository_test.dart:88]: Should we test that the `fromJson` method creates an instance from the JSON-encodable map?

File: metrics/ci_integrations/test/client/github_actions/models/github_user_test.dart [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      () {
+        const githubUser = GithubUser(id: id, login: login);
+
+        expect(githubUser.id, equals(id));
+        expect(githubUser.login, equals(login));
+      },
+    );
+
+    test(
+      ".fromJson() returns null if the given json is null",
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/github_actions/models/github_user_test.dart:30]: Should we test that the `fromJson` creates an instance from the given JSON?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Well done, @solid-maxim & @solid-tetianamovlian! I've added a couple of suggestions here - please take a look. Overall, the code looks clean!

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-maxim & @solid-tetianamovlian!
